### PR TITLE
Update infinera-groove.inc.php

### PR DESCRIPTION
**Pull request by thundersin**
Adding Transmit optical power to dbm sensor for infinera grooves

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

image

### CODE REVIEW

**Review by laf — DISMISSED** [review]
LGTM. Test data already exists for snmpsim but just needs the json data.

@thundersin Can you run then include the changed json file:

`./scripts/save-test-data.php -o infinera-groove -m sensors`

**Comment by murrant** [comment]
@thundersin this has been fixed to properly reference python3.  Update and you should be able to run save-test-data.php

**Comment by thundersin** [comment]
> @thundersin this has been fixed to properly reference python3. Update and you should be able to run save-test-data.php

@murrant Same error unfortunately:

[09:19 Mon Oct 07] librenms@librenms ~1 âž lnms --version
LibreNMS 24.9.0
[09:22 Mon Oct 07] librenms@librenms ~âž ./daily.sh
Fetching new release information                   OK
Updating to latest release                         OK
Updating Composer packages                         OK
Updating SQL-Schema                                OK
Updating submodules                                OK
Cleaning up DB                                     OK
Fetching notifications                             OK
Caching PeeringDB data                             OK
[09:23 Mon Oct 07] librenms@librenms ~âž ./scripts/save-test-data.php -o infinera-groove -m sensors
sh: 1: exec: python: not found
sh: 1: exec: /opt/librenms/.python_venvs/snmpsim//bin/pip: not found
Waiting for snmpsim to initialize...
Failed to start snmpsim, make sure it is installed, working, and there are no bad snmprec files.
Run ./scripts/save-test-data.php --snmpsim to see the log output

**Comment by laf** [comment]
If you are on the monthly release version then it won't be available until 24.10.0.

You could temporarily add the patch and remove afterwards:

`./scripts/github-apply 16492`

Then run the command to generate the data and then:

`./scripts/github-remove -s`

**Comment by thundersin** [comment]
@laf Thank you for your help. All done now.

File: includes/discovery/sensors/dbm/infinera-groove.inc.php [hunk 1]
```diff
@@ -29,4 +29,10 @@
         $value = $data['portRxOpticalPower'];
         discover_sensor(null, 'dbm', $device, $oid, 'portRxOpticalPower.' . $index, 'infinera-groove', $descr, null, '1', null, null, null, null, $value);
     }
+    if (is_numeric($data['portTxOpticalPower']) && $data['portTxOpticalPower'] != -99) {
```
Comment (laf) [inline includes/discovery/sensors/dbm/infinera-groove.inc.php:32]: Looks like Receive Power is available in the MIB as well, can you add that in?

**Review by laf — CHANGES_REQUESTED** [review]

File: includes/discovery/sensors/dbm/infinera-groove.inc.php (hunk 1 above)
Comment (thundersin) [reply includes/discovery/sensors/dbm/infinera-groove.inc.php:32]: > Looks like Receive Power is available in the MIB as well, can you add that in?

Receive power is already there my friend. I am simply adding the transmit power.

**Review by laf — APPROVED** [review]
LGTM

**Comment by laf** [comment]
Haha, so it is. Thanks for contributing :)

**Pull request merged**